bin/main.rb

```diff
@@ -1,2 +1,123 @@
+board = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+# Display Current Board
+puts 'Current Board:', "\n"
+print board[0], '  |  ', board[1], '  |  ', board[2], "\n"
+print "\n"
+print board[3], '  |  ', board[4], '  |  ', board[5], "\n"
+print "\n"
+print board[6], '  |  ', board[7], '  |  ', board[8], "\n"
+print "\n"
+# Initialize Control Variables
+played_one = false
+played_two = true
+count = 0
+check_input = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+game_over = false
+# Read players' optionn from the keyboard,
+# Validate input,
+# Update Game Board,
+# Display Current Board
+# Update Control Variables
+
+while !game_over && count < 9 && !check_input.empty?
+  # unless check_input.empty?
+  while !played_one && played_two && count < 9
+    puts "#{player_one_name}'s turn Please choose between #{check_input} count: #{count}"
+    player_one_choice = gets.to_i # Read player option from the user
+    next unless check_input.include?(player_one_choice)
+
+    board[player_one_choice - 1] = input_symb1 # unless !check_input.include?(player_one)
+    check_input.delete(player_one_choice)
+    # display_board(current_board)
+    puts 'Current Board:', "\n"
+    print board[0], '  |  ', board[1], '  |  ', board[2], "\n"
+    print "\n"
+    print board[3], '  |  ', board[4], '  |  ', board[5], "\n"
+    print "\n"
+    print board[6], '  |  ', board[7], '  |  ', board[8], "\n"
+    print "\n"
+    i = player_one_choice - 1
+    game_over = [[0, 1, 2], [3, 4, 5], [6, 7, 8], [0, 3, 6], [1, 4, 7], [2, 5, 8], [0, 4, 8], [2, 4, 6]].any? do |a|
+      a.include?(i) && a.all? { |x| board[x] == board[i] }
+    end
+    if game_over
+      puts "Game over !! Congratulation #{player_one_name}, you are the winner", "\n"
+      break
+    end
+
+    count += 1
+    played_one = true
+    played_two = false
+  end
+
+  while !played_two && played_one && count < 9
+    puts "#{player_two_name}'s turn Please choose between #{check_input} count: #{count}"
+    player_two_choice = gets.to_i # Read player option from the user
+    next unless check_input.include?(player_two_choice) # Checks input, validate or reject invalid input
+
+    board[player_two_choice - 1] = input_symb2 # unless !check_input.include?(player_one)
+    check_input.delete(player_two_choice)
+    # display_board(current_board)
+    puts 'Current Board:', "\n"
+    print board[0], '  |  ', board[1], '  |  ', board[2], "\n"
+    print "\n"
+    print board[3], '  |  ', board[4], '  |  ', board[5], "\n"
+    print "\n"
+    print board[6], '  |  ', board[7], '  |  ', board[8], "\n"
+    print "\n"
+    i = player_two_choice - 1
+    game_over = [[0, 1, 2], [3, 4, 5], [6, 7, 8], [0, 3, 6], [1, 4, 7], [2, 5, 8], [0, 4, 8], [2, 4, 6]].any? do |a|
+      a.include?(i) && a.all? { |x| board[x] == board[i] }
+    end
+    if game_over
+      puts "Game over !! Congratulation #{player_two_name}, you are the winner", "\n"
+      break
+    end
+    count += 1
+    played_one = false
+    played_two = true
+  end
```

Comment:
This is optional for **this milestone** but must be taken in consideration in the next milestone. Think about using a **switch_player** method and no inner while loop for each player.

Reply:
Ok, thanks! I will work on them.
Kind regards